homework/queries.sql

```diff
@@ -1,19 +1,114 @@
---- Get the top 3 product types that have proven most profitable
+-- Connect to the star_db, needed for executing queries
+\c star_db
 
+-- Get the top 3 product types that have proven most profitable
+SELECT p.product_name,
+  p.product_code,
+  r.profits
+FROM dim_products AS p
+INNER JOIN (
+  SELECT DISTINCT(product_code),
+      profits
+  FROM  fact_order_items
+  ORDER BY profits DESC
+  LIMIT 3
+) AS r
+USING (product_code);
 
---- Get the top 3 products by most items sold
+-- Get the top 3 products by most items sold
+SELECT p.product_name,
+  p.product_code,
+  t.total_quantity
+FROM dim_products AS p
+INNER JOIN (
+  SELECT product_code,
+    SUM(quantity_ordered) AS total_quantity
+    FROM fact_order_items
+    GROUP BY product_code
+    ORDER BY total_quantity DESC
+    LIMIT 3) AS t
+  USING(product_code);
 
+-- Get the top 3 products by items sold per country of customer for: USA,
+-- Spain, Belgium
+SELECT s.product_code,
+      p.product_name,
+      s.country,
+      s.total_quantity,
+      s.rank
+FROM (
+  SELECT s.product_code,
+    s.country,
+    s.total_quantity,
+    RANK() OVER (PARTITION BY s.country ORDER BY s.total_quantity DESC)
+  FROM (
+    SELECT f.product_code,
+      SUM(f.quantity_ordered) AS total_quantity,
+      c.country
+    FROM fact_order_items AS f
+    INNER JOIN dim_customers AS c
+      USING(customer_number)
+    WHERE c.country IN ('USA', 'Spain', 'Belgium')
+    GROUP BY (f.product_code, c.country)
+    ORDER BY total_quantity DESC) AS s) AS s
+INNER JOIN dim_products AS p
+  USING(product_code)
+WHERE s.rank <= 3
+ORDER BY s.country, s.rank;
 
---- Get the top 3 products by items sold per country of customer for: USA, Spain, Belgium
+-- Get the most profitable day of the week
+SELECT d.day_of_week,
+  SUM(f.profits) AS total_per_day
+FROM fact_order_items AS f
+INNER JOIN dim_dates AS d
+  ON order_date = iso_date
+GROUP BY d.day_of_week
+ORDER BY total_per_day DESC
+LIMIT 1;
 
+-- Get the top 3 city-quarters with the highest average profit margin in their
+-- sales
+SELECT o.city,
+  CAST(AVG(f.profit_margin) AS NUMERIC(10,2)) AS avg_profit_margin,
```

Comment:
Nice, specific delivery format!

Reply:
(I mean, I would generally prefer the visualization layer to do that, but assuming your output is your visualization, it's a nice touch!)